core-common/src/main/java/org/zalando/nakadi/filters/TracingFilter.java

```diff
@@ -67,16 +110,39 @@ protected void doFilterInternal(final HttpServletRequest request,
         try (Closeable ignored = TracingService.activateSpan(span)) {
             span.setTag("client_id", authorizationService.getSubject().map(Subject::getName).orElse("-"));
 
-            //execute request
             filterChain.doFilter(request, response);
 
+            if (request.isAsyncStarted()) {
+                request.getAsyncContext().addListener(new AsyncRequestSpanFinalizer(span, request, response));
+            }
```

Comment:
I suggest to move it below (see suggestion), so the finalization process will not be spread all over the method, but will concentrate in a single place.